### PR TITLE
Migrate to gatsby-source-strapi v1

### PR DESCRIPTION
**Pull request by remidej**
Hello!

Here's what this PR does:

- Migrate the starter to gatsby-source-strapi
- Remove the gatsby-plugin-graphql-image, since we're not using graphql anymore
- Attempts to fix secret mode issues

Here are the issues that I know about it:

- All images from contentSections are not handled by GatsbyImage (so most images on the starter are dumb  tags). But this problem is a gatsby-source-plugin issue (media fields are not parsed when they're inside dynamic zones), so I don't think we need to fix it in this starter. However, it will be a priority for the next major gatsby-plugin-strapi release.
- Preview mode requires you to set the locale prefix in the URL, even though most other pages don't. I didn't remember how we handled the locale prefix for normal pages in the end, so I'll let you look at this one Mark
- Clicking on the Strapi navbar logo from a preview mode page goes to "/:locale/preview/", which is a 404

### CODE REVIEW

File: starter/src/templates/page.js [hunk 1]
```diff
@@ -21,291 +21,103 @@ const DynamicPage = ({ data, pageContext }) => {
...
-                gatsbyImageData(
-                  placeholder: BLURRED
-                  formats: [AUTO, WEBP, AVIF]
-                )
-              }
-            }
-          }
-          smallText
-        }
+  fragment GlobalData on StrapiGlobal {
```
Comment (markkaylor) [inline starter/src/templates/page.js:24]: Nice, good idea to use the fragment 👍

**Comment by markkaylor** [comment]
Hey looks good to me @remidej .  I opened #21 on your branch to fix preview mode. The images are working on that branch 🤷‍♂️ .  I rebased main on to my branch without thinking 🤦.   I think if you do the same here it will resolve the conflicts on my branch.

**Comment by markkaylor** [comment]
Something strange going on with the image here: 
Screenshot 2021-07-21 at 15 32 59

**Comment by remidej** [comment]
I found that around tablet sizes, the image was indeed stretched. I tried changing the media queries, does it work for you?

**Comment by markkaylor** [comment]
Seems to be working fine in Firefox 🙃  but still stretched in chrome 😢 

firefox:
Screenshot 2021-07-21 at 16 38 11

chrome:
Screenshot 2021-07-21 at 16 38 17

**Comment by remidej** [comment]
Thanks, I hadn't tested it in Chrome. Can you try again?

**Review by markkaylor — APPROVED** [review]
Awesome! Looks good to me! 🚀

**Pull request merged**